File: hangman.py
import random

words = ["apple", "computer", "python", "school", "football"]

word = random.choice(words)

guessed = []
wrong_guesses = 0
max_wrong = 6

print("===== HANGMAN GAME =====")

while wrong_guesses < max_wrong:

    # Display word
    display = ""

    for letter in word:
        if letter in guessed:
            display += letter + " "
        else:
            display += "_ "

    print("\nWord:", display)
    print("Wrong guesses:", wrong_guesses, "/", max_wrong)

    # Check win
    if all(letter in guessed for letter in word):
        print("\n YOU WON!")
        print("The word was:", word)
        break

    guess = input("Enter a letter: ").lower()

    # Validate input
    if len(guess) != 1 or not guess.isalpha():
        print("Please enter ONE letter.")
        continue

    if guess in guessed:
        print("You already guessed this letter!")
        continue

    guessed.append(guess)

    if guess in word:
        print("Correct!")
    else:
        print("Wrong!")
        wrong_guesses += 1

else:
    print("\n GAME OVER!")
    print("The word was:", word)

File: Hangman.java
import java.util.*;

public class Hangman {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        String[] words = {
            "apple",
            "computer",
            "java",
            "school",
            "football"
        };

        Random random = new Random();

        String word = words[random.nextInt(words.length)];

        ArrayList<Character> guessed = new ArrayList<>();

        int wrongGuesses = 0;
        int maxWrong = 6;

        System.out.println("===== HANGMAN GAME =====");

        while (wrongGuesses < maxWrong) {

            // Display word
            System.out.print("\nWord: ");

            for (int i = 0; i < word.length(); i++) {

                char letter = word.charAt(i);

                if (guessed.contains(letter)) {
                    System.out.print(letter + " ");
                } else {
                    System.out.print("_ ");
                }
            }

            System.out.println();
            System.out.println("Wrong guesses: "
                    + wrongGuesses + "/" + maxWrong);

            // Check win
            boolean won = true;

            for (int i = 0; i < word.length(); i++) {

                if (!guessed.contains(word.charAt(i))) {
                    won = false;
                    break;
                }
            }

            if (won) {
                System.out.println("\n🎉 YOU WON!");
                System.out.println("The word was: " + word);
                break;
            }

            // Take guess
            System.out.print("Enter a letter: ");
            char guess = sc.next().toLowerCase().charAt(0);

            // Already guessed
            if (guessed.contains(guess)) {
                System.out.println("You already guessed this letter!");
                continue;
            }

            guessed.add(guess);

            // Check letter
            if (word.indexOf(guess) != -1) {
                System.out.println("✅ Correct!");
            } else {
                System.out.println("❌ Wrong!");
                wrongGuesses++;
            }
        }

        if (wrongGuesses == maxWrong) {
            System.out.println("\n GAME OVER!");
            System.out.println("The word was: " + word);
        }

        sc.close();
    }
}
